docs/kubernetes-build-win-binaries.md

```diff
@@ -1,87 +1,29 @@
-
-6. Build windows kubelet.exe
-   ```
-   cd $HOME
-   git clone https://github.com/kubernetes/kubernetes $GOPATH/src/k8s.io/kubernetes
-   cd $GOPATH/src/k8s.io/kubernetes
-   make WHAT=cmd/kubelet
-   make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=windows/amd64
-   ```
-7. Build windows kube-proxy.exe
-   ```
-   cd $GOPATH/src/k8s.io/kubernetes
-   make WHAT=cmd/kube-proxy
-   make WHAT=cmd/kube-proxy KUBE_BUILD_PLATFORMS=windows/amd64
-   ```
-
-8. Build the zip file
-   ```
-   cd $HOME
-   rm -rf uploadk
-   mkdir uploadk
-   cd uploadk
-   mkdir k
-   cp $HOME/kubelet/kubernetes/_output/local/bin/windows/amd64/kubelet.exe k
-   cp $HOME/kube-proxy/kubernetes/_output/local/bin/windows/amd64/kube-proxy.exe k
-   wget https://storage.googleapis.com/kubernetes-release/release/v1.4.6/bin/windows/amd64/kubectl.exe -P k
-   chmod 775 k/kubectl.exe
-   cat <<EOT >> k/Dockerfile
-   FROM microsoft/windowsservercore
-   
-   ADD pause.ps1 /pause/pause.ps1
-   	
-   CMD powershell /pause/pause.ps1
-   EOT
-   cat <<EOT >> k/pause.ps1
-   while(\$true)
-   {
-       Start-Sleep -Seconds 60
-   }
-   EOT
-   wget https://nssm.cc/release/nssm-2.24.zip
-   unzip nssm-2.24.zip
-   cp nssm-2.24/win64/nssm.exe k
-   rm -rf nssm-2.24*
-   chmod 775 k/nssm.exe
-   zip -r k.zip *
-   ```
-
-9. Upload to storage.  This assumes you have az cli installed and you have a storage account
-   ```
-   export AZURE_STORAGE_CONNECTION_STRING="DefaultEndpointsProtocol=https;AccountName=enter_your_account;AccountKey=enter_your_key"
-   azure storage blob upload k.zip k8scontainer k.zip -q
-   ```
+# Building Windows Kubernetes Binaries and deploy to an Azure storage account
+
+The following instructions show how to deploy the Windows Kubernetes Binaries and deploy them to an Azure Storage Account.
+
+### Prerequisites
+* Azure Storage Account and Azure Storage Container to store Windows binaries
+* Access to [winnat.sys](https://blogs.technet.microsoft.com/virtualization/2016/05/25/windows-nat-winnat-capabilities-and-limitations/) stored in a storage container; at present Microsoft is not able to distribute winnat.sys.
+* Go development environment for cross-building
+
+### Set Azure Storage credentials and Container name
+```
+$ export AZURE_STORAGE_CONNECTION_STRING="DefaultEndpointsProtocol=https;AccountName=MyStorageAccountName;AccountKey=..." \
+AZURE_STORAGE_CONTAINER_NAME=MyStorageContainerName
+```
+
+### Run `make build-windows-k8s`
+
+The `make build-windows-k8s` will do the following:
+- Clone the fork of Azure/kubernetes which include Windows fixes not yet in upstream Kubernetes
+- Build kubelet.exe, kube-proxy.exe
+- Download kubectl.exe for desired release
+- Download [NSSM](https://nssm.cc) which is used to start the kubelet on Windows
+- Download [Windows NAT](https://blogs.technet.microsoft.com/virtualization/2016/05/25/windows-nat-winnat-capabilities-and-limitations/) (WinNAT) used to provide required NAT networking functionality for Windows containers
```

Comment:
Where is the URL for downloading winnat? Did we upload somewhere?

Reply:
I uploaded to the ACS storage container.